discovery-provider/src/tasks/index.py

```diff
@@ -340,366 +374,298 @@ def update_ursm_address(self):
             address=registry_address, abi=abi_values["Registry"]["abi"]
         )
         user_replica_set_manager_address = registry_instance.functions.getContract(
-            bytes("UserReplicaSetManager", "utf-8")
+            bytes(USER_REPLICA_SET_MANAGER_CONTRACT, "utf-8")
         ).call()
         if user_replica_set_manager_address != zero_address:
-            get_contract_addresses()[
-                "user_replica_set_manager"
-            ] = web3.toChecksumAddress(user_replica_set_manager_address)
+            get_contract_addresses()[USER_REPLICA_SET_MANAGER] = web3.toChecksumAddress(
+                user_replica_set_manager_address
+            )
             logger.info(
                 f"index.py | Updated user_replica_set_manager_address={user_replica_set_manager_address}"
             )
 
 
-def save_and_get_skip_tx_hash(session, redis):
+def get_tx_hash_to_skip(session, redis):
```

Comment:
don't think you need session here

Reply:
oof i had to add this because it makes the test work ... let me make it work without

Reply:
this is kind of tricky; do you mind if i add as a fast follow?